File: VHBoomMenuButtonDemo/FadeViewsVC.m
//
//  FadeViewsVC.m
//  VHBoomMenuButtonDemo
//
//  Created by viktorhuang on 2017/5/7.
//  Copyright © 2017年 Nightonke. All rights reserved.
//

#import "FadeViewsVC.h"
#import "BuilderManager.h"
#import <BoomMenuButton/BoomMenuButton.h>

@interface FadeViewsVC ()

@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb1;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb2;

@end

@implementation FadeViewsVC

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    [self initializeBmb1];
    [self initializeBmb2];
}

- (void)initializeBmb1
{
    self.bmb1.buttonEnum = VHButtonSimpleCircle;
    self.bmb1.piecePlaceEnum = VHPiecePlaceDOT_9_1;
    self.bmb1.buttonPlaceEnum = VHButtonPlaceSC_9_1;
    for (int i = 0; i < self.bmb1.pieceNumber; i++)
    {
        [self.bmb1 addBuilder:[BuilderManager simpleCircleButtonBuilder]];
    }
    self.bmb1.dotRadius = 0;
    self.bmb1.draggable = YES;
    
    UIImageView *imageView = [[UIImageView alloc] initWithFrame:CGRectMake(10, 10, 40, 40)];
    imageView.image = [UIImage imageNamed:[BuilderManager imageName]];
    [self.bmb1 addFadeView:imageView];
}

- (void)initializeBmb2
{
    self.bmb2.buttonEnum = VHButtonHam;
    self.bmb2.piecePlaceEnum = VHPiecePlaceHAM_4;
    self.bmb2.buttonPlaceEnum = VHButtonPlaceHAM_4;
    for (int i = 0; i < self.bmb2.pieceNumber; i++)
    {
        [self.bmb2 addBuilder:[BuilderManager hamButtonBuilderWithDifferentPieceColor]];
    }
    self.bmb2.hamWidth = 0;
    self.bmb2.hamHeight = 0;
    self.bmb2.draggable = YES;
    
    UIImageView *imageView = [[UIImageView alloc] initWithFrame:CGRectMake(10, 8, 40, 40)];
    imageView.image = [UIImage imageNamed:[BuilderManager imageName]];
    [self.bmb2 addFadeView:imageView];
    
    UILabel *label = [[UILabel alloc] initWithFrame:CGRectMake(10, 42, 40, 10)];
    label.font = [UIFont systemFontOfSize:8];
    label.text = @"BOOM!";
    label.textColor = UIColor.whiteColor;
    label.textAlignment = NSTextAlignmentCenter;
    [self.bmb2 addFadeView:label];
}

@end

File: VHBoomMenuButtonSwiftDemo/FadeViewsVC.swift
//
//  FadeViewsVC.swift
//  VHBoomMenuButtonSwiftDemo
//
//  Created by Nightonke on 2017/5/8.
//  Copyright © 2017 Nightonke. All rights reserved.
//

import UIKit
import BoomMenuButton

class FadeViewsVC: BoomMenuButtonVC {

    @IBOutlet weak var bmb1: BoomMenuButton!
    @IBOutlet weak var bmb2: BoomMenuButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        initializeBmb1()
        initializeBmb2()
    }
    
    func initializeBmb1() {
        bmb1.buttonEnum = .simpleCircle
        bmb1.piecePlaceEnum = .dot_9_1
        bmb1.buttonPlaceEnum = .sc_9_1
        for _ in 0..<bmb1.piecePlaceEnum.pieceNumber() {
            bmb1.addBuilder(BuilderManager.simpleCircleButtonBuilder())
        }
        bmb1.dotRadius = 0
        bmb1.draggable = true
        
        let imageView: UIImageView = UIImageView.init(frame: CGRect.init(x: 10, y: 10, width: 40, height: 40))
        imageView.image = UIImage.init(named: BuilderManager.imageName())
        bmb1.addFadeView(imageView)
    }
    
    func initializeBmb2() {
        bmb2.buttonEnum = .ham
        bmb2.piecePlaceEnum = .ham_4
        bmb2.buttonPlaceEnum = .ham_4
        for _ in 0..<bmb2.piecePlaceEnum.pieceNumber() {
            bmb2.addBuilder(BuilderManager.hamButtonBuilderWithDifferentPieceColor())
        }
        bmb2.hamWidth = 0
        bmb2.hamHeight = 0
        bmb2.draggable = true
        
        let imageView: UIImageView = UIImageView.init(frame: CGRect.init(x: 10, y: 8, width: 40, height: 40))
        imageView.image = UIImage.init(named: BuilderManager.imageName())
        bmb2.addFadeView(imageView)
        
        let label: UILabel = UILabel.init(frame: CGRect.init(x: 10, y: 42, width: 40, height: 10))
        label.font = UIFont.systemFont(ofSize: 8)
        label.text = "BOOM!"
        label.textColor = UIColor.white
        label.textAlignment = NSTextAlignment.center
        bmb2.addFadeView(label)
    }
}
